llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -3267,11 +3267,23 @@ void VPPredInstPHIRecipe::execute(VPTransformState &State) {
   // also do that packing, thereby "hoisting" the insert-element sequence.
   // Otherwise, a phi node for the scalar value is needed.
   if (State.hasVectorValue(getOperand(0))) {
-    Value *VectorValue = State.get(getOperand(0));
-    InsertElementInst *IEI = cast<InsertElementInst>(VectorValue);
-    PHINode *VPhi = State.Builder.CreatePHI(IEI->getType(), 2);
-    VPhi->addIncoming(IEI->getOperand(0), PredicatingBB); // Unmodified vector.
-    VPhi->addIncoming(IEI, PredicatedBB); // New vector with inserted element.
+    auto *VecI = cast<Instruction>(State.get(getOperand(0)));
+    assert(isa<InsertElementInst>(VecI) ||
+           isa<InsertValueInst>(VecI) &&
+               "Packed recipes must generate an insertelement or insertvalue");
```

Comment:
```suggestion
               "Packed operands must generate an insertelement or insertvalue");
```

to match the code looking up the operand?